packages/@react-spectrum/provider/src/mediaQueries.ts

```diff
@@ -59,3 +61,54 @@ export function useScale(theme: Theme): Scale {
 
   return 'medium';
 }
+
+export function useBreakpoint(breakpoints: Breakpoints) {
+  // sort breakpoints in ascending order.
+  let entries = Object.entries(breakpoints).sort(([, valueA], [, valueB]) => valueA - valueB);
+  let breakpointQueries = entries.map(([, value], index) => {
+    if (index === entries.length - 1) {
+      return `(min-width: ${value}px)`;
+    } else {
+      return `(min-width: ${value}px) and (max-width: ${entries[index + 1][1]}px)`;
```

Comment:
since we're sorted do we want to upper bound each range?
what happens if someone only supplies {S, L} to a component, but {S, M, L} is supported by the application? I guess, do we expect it to be S or L in the M case? or would we expect it to be base? or do we require that if someone uses this syntax, that they provide all? in which case, should we emit a warning or throw?

I see you've opted for base to be used. This isn't a criticism, just trying to figure out what people would expect. We'll need to document it.

Reply:
Also, I think this will create
`[380, 768], [768, 1024], [1024, INF]`
do we need to worry about `[0, 380]`? or should that be base as well?

apologies if i've misinterpreted the code, I haven't had a chance to run it yet

Reply:
I think we should probably target `min/max-device-width` so that react spectrum inside iframes doesn't try to use lower breakpoints than expected

Reply:
> since we're sorted do we want to upper bound each range?

Oh, as you said we may not need upper bound anymore.

> what happens if someone only supplies {S, L} to a component, but {S, M, L} is supported by the application?

Current implementation leads to use base values, but returning S in the M case also makes sense to me.
What is your opinion?

Reply:
> do we need to worry about [0, 380]? or should that be base as well?

I thought it should be base as well.
If you think this is too implicit, adding `XS: 0` or `min: 0` or `base: 0` to the breakpoints might be a possible solution?

> I think we should probably target min/max-device-width so that react spectrum inside iframes doesn't try to use lower breakpoints than expected (haven't tested this to see if it's a concern)

👍

Reply:
>  returning S in the M case also makes sense to me

Other libraries like tailwind or chakra-ui were following this rule and I decided to follow them as well.
For different use-cases, I added `mobileFirst` props temporarily.

Reply:
> I think device-width is deprecated
Ah, yes, you are correct, I had missed that. I tried out the code though, and it currently is responding to iframe size. I don't have another idea for dealing with it right now, nor do I think it should block this. Something to think about though.

> I added mobileFirst props temporarily.
This is an interesting idea. I'll play with the storybook.